app/main.py

```diff
@@ -1 +1,20 @@
-# write your code here
+import app.errors as err
+from app.cafe import Cafe
+
+
+def go_to_cafe(cafe: Cafe, friends: list):
+    mask_to_buy = 0
+    vacs_group = True
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except err.VaccineError:
+            vacs_group = False
```

Comment:
In the task it is mentioned "If at least one of the friend has problems with vaccines the function should return a message `"All friends should be vaccinated"." So, you don't need to create vacs_group = True, then change if to False and then make verification the line 16-17, you just need to find out first such person and return the message

Reply:
I agree that in this task it`s possible to do such thing, but in general this will produce "spaghetti" code. Also, this code doesn`t meant to be as fast as possible, so it`s allowed to go through whole list every time.

Reply:
I think the suggestion above won't really lead to the complicated code. It can be even more readable because you will show explicitly that` VaccineError` lead to a specific returned value